Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -281,41 +342,49 @@ - (void)resetUpcomingSampleBufferPresentationTimeExpectations;
 {
     assertIsMainThread();
 
-    ASSERT(m_compressedSampleQueue);
-
-    if (!m_compressedSampleQueue)
-        return;
-
-    PAL::CMBufferQueueReset(m_compressedSampleQueue.get());
-    PAL::CMTimebaseSetTimerDispatchSourceNextFireTime(m_timebase.get(), m_timerSource.get(), PAL::kCMTimeInvalid, 0);
+    ++m_flushId;
```

Comment:
Ah I see, m_flushId is called at the same time decompression session flush is called.
The scheme I was mentioning above would require flushing the decompressions session from the work queue.

There might still be room for simplification but I am not exactly sure how then.

Reply:
yes, the flushId is a scheme I've used quite a bit in similar case. It's simple, avoid the need for sync dispatch nor the use of a lock.

And it makes it much easier to fully dismiss any in-flight operation where we can dismiss the result.

We could have achieved similar cancellation of inflight decoding by using a NativePromise::Request , but this ended up the need of a lock which I wanted to avoid.